gluecodium/src/main/resources/templates/ffi/FfiCallbacksQueueHeader.mustache

```diff
@@ -0,0 +1,129 @@
+{{!!
+  !
+  ! Copyright (C) 2016-2020 HERE Europe B.V.
+  !
+  ! Licensed under the Apache License, Version 2.0 (the "License");
+  ! you may not use this file except in compliance with the License.
+  ! You may obtain a copy of the License at
+  !
+  !     http://www.apache.org/licenses/LICENSE-2.0
+  !
+  ! Unless required by applicable law or agreed to in writing, software
+  ! distributed under the License is distributed on an "AS IS" BASIS,
+  ! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ! See the License for the specific language governing permissions and
+  ! limitations under the License.
+  !
+  ! SPDX-License-Identifier: Apache-2.0
+  ! License-Filename: LICENSE
+  !
+  !}}
+{{>ffi/FfiCopyrightHeader}}
+
+#pragma once
+
+#include <functional>
+#include <future>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <unordered_map>
+
+{{#internalNamespace}}
+namespace {{.}}
+{
+{{/internalNamespace}}
+namespace ffi
+{
+/**
+ * A queue that manages a list of callbacks.
+ */
+class CallbackQueue
+{
+private:
+    struct Entry
+    {
+        std::function<void()> func;
+        std::promise<bool> done;
+    };
+
+    std::mutex mutex;
+    std::queue<Entry> queue;
+
+    bool closed;
```

Comment:
Initialise 'closed' and 'incoming' right here in place and remove ctor.
Also possibly must be m_is_closed and m_has_incoming

Reply:
Done